stats.js

```diff
@@ -46,52 +91,56 @@ function stats(opts) {
       return;
     }
 
-    var stream = nes(container.stats.bind(container));
+    if (streamMode) {
+      var stream = nes(container.stats.bind(container));
 
-    streams[data.Id] = stream;
+      streams[data.Id] = stream;
 
-    var previousSystem = 0;
-    var previousCpu = 0;
+      var previousSystem = 0;
+      var previousCpu = 0;
 
-    var sampleCount = 0;
-    var cpuSum = 0;
-    var sysSum = 0;
+      var sampleCount = 0;
+      var cpuSum = 0;
+      var sysSum = 0;
 
-    pump(
-      stream,
-      split(JSON.parse),
-      through.obj(function(stats, enc, cb) {
-        sampleCount++
+      pump(
+        stream,
+        split(JSON.parse),
+        through.obj(function(stats, enc, cb) {
+          sampleCount++
 
-        cpuSum += stats.cpu_stats.cpu_usage.total_usage
-        sysSum += stats.cpu_stats.system_cpu_usage
+          cpuSum += stats.cpu_stats.cpu_usage.total_usage
+          sysSum += stats.cpu_stats.system_cpu_usage
 
-        if (sampleCount >= interval) {
-          stats.cpu_stats.cpu_usage.total_usage = cpuSum/sampleCount;
-          stats.cpu_stats.system_cpu_usage = sysSum/sampleCount;
+          if (sampleCount >= interval) {
+            stats.cpu_stats.cpu_usage.total_usage = cpuSum/sampleCount;
+            stats.cpu_stats.system_cpu_usage = sysSum/sampleCount;
 
-          var percent = calculateCPUPercent(stats, previousCpu, previousSystem)
-          stats.cpu_stats.cpu_usage.cpu_percent = percent
+            var percent = calculateCPUPercent(stats, previousCpu, previousSystem)
+            stats.cpu_stats.cpu_usage.cpu_percent = percent
```

Comment:
The current stats also calculates the CPU stats, can you add those as well for the polling system? Maybe you can extract these into a function.